Ensure the correct Rust version when running locally and on CI.

cyplo opened the pull request:
* As we encourage the scripts to be runnable locally as well as on CI -
renamed the `ci` directory to `scripts`
* Install script should be safer to run locally now - it only sets the
default global compiler version on CI
* Each script sets its own rustup override so that we make sure the
build and tests are run with the correct Rust version

zrzka commented on .travis.yml:
```diff
@@ -1,14 +1,14 @@
 
 install:
-  - ci/install.sh
+  - scripts/install.sh
```
Can't say that I like it, but ...

cyplo replied:
Any suggestions ? My thinking was that these are not really `ci` specific anymore, as you can run them locally etc. and looking at a sample balena project (rendition), they use `scripts` as well for this.

zrzka replied:
Nah, I'm fine with it, just a nitpick.

zrzka commented on scripts/build-wasm.sh:
```diff
@@ -13,6 +13,9 @@ source "${HOME}/.nvm/nvm.sh"
 source "${HOME}/.cargo/env"
 
+echo "Setting rustup override for this project"
+rustup override set `cat rust-toolchain`
```
Please, don't use backticks. Use `$()` as it is POSIX & can be nested (copy & pasted freely around).

zrzka commented on scripts/deploy.sh:
```diff
@@ -2,10 +2,15 @@
 nvm use
 
+echo "Setting rustup override for this project"
+rustup override set `cat rust-toolchain`
```
Please, don't use backticks. Use `$()` as it is POSIX & can be nested (copy & pasted freely around). And be consistent with `HERE="$...`.

zrzka commented on scripts/deploy.sh:
```diff
@@ -14,7 +19,7 @@ cargo publish
 echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
 
-ci/build-wasm.sh
+$HERE/build-wasm.sh
```
Will fail if `$HERE` will contain space for example. Use `"${HERE}/build-wasm.sh"`.

zrzka commented on scripts/install.sh:
```diff
@@ -3,8 +3,15 @@
 set -e
 set -o pipefail
 
+rustup_commandline="curl https://sh.rustup.rs -sSf | sh -s -- -y"
```
Seems that `rustup_commandline` is better, `line` suffix has no meaning here.

We should use `RUSTUP_COMMAND` or keep it lowercased `rustup_command`, but then all other variables (script local) should be lowercased as well (like `HERE`, ...).

zrzka commented on scripts/test.sh:
```diff
@@ -28,24 +33,23 @@ cargo test
 cargo package ${CARGO_PACKAGE_ARGS}
 
-ci/build-wasm.sh
+$HERE/build-wasm.sh
```
Same as previous comment, space in `HERE`.

zrzka commented on scripts/deploy.sh:
```diff
@@ -14,7 +19,7 @@ cargo publish
 echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
 
-ci/build-wasm.sh
+"$HERE/build-wasm.sh"
```
`"${HERE}/..." to be consistent.

zrzka commented on scripts/install.sh:
```diff
@@ -3,8 +3,15 @@
+fi
 echo "Installing Rust toolchain..."
-curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain $(cat rust-toolchain)
+eval $RUSTUP_COMMAND
```
Not sure if this one should be quoted as well?

cyplo replied:
It's a good question and I'm not sure as well - all the examples I found on the internet were unquoted - but haven't found anything definitive. This definitely works now and it contains spaces already, so maybe eval does not care ? I will change it to quoted one for consistency though as it seems to work as well.

zrzka replied:
If it works and it's ok, we can keep it. Was just asking, not sure here ...

zrzka commented on scripts/test.sh:
```diff
@@ -28,24 +33,23 @@ cargo test
 cargo package ${CARGO_PACKAGE_ARGS}
 
-ci/build-wasm.sh
+"$HERE/build-wasm.sh"
```
`"${HERE}/..." to be consistent.

zrzka commented on scripts/test.sh:
```diff
@@ -28,24 +33,23 @@ cargo test
 echo "Trying to package Rust crate..."
 CARGO_PACKAGE_ARGS=''
-if ! [ "$CI" = true ]; then
+if [ -z "$CI" ]; then
```
`"${CI}" to be consistent.

zrzka commented:
When we are at it, can we add the `#[allow(clippy)]` to `tests.rs`? Because all these changes are here because of the `clippy` ...

zrzka commented on scripts/install.sh:
```diff
@@ -3,8 +3,15 @@
 
+RUSTUP_COMMAND="curl https://sh.rustup.rs -sSf | sh -s -- -y"
+if [ ! -z $CI ]; then
+    default_compiler=`cat rust-toolchain`
```
Upper case :)

cyplo commented on scripts/install.sh:
```diff
@@ -3,8 +3,15 @@
+if [ ! -z $CI ]; then
+    default_compiler=`cat rust-toolchain`
+    echo "Setting the default compiler to ${default_compiler}"
+    RUSTUP_COMMAND="$RUSTUP_COMMAND --default-toolchain ${default_compiler}"
```
@zrzka do you think `default_compiler` should be uppercase as well ? I seen "local" variables being lowercase in scripts before - but happy to change to uppercase, if that's more consistent ?

zrzka replied:
@cyplo shell rule is that upper cased are envs & builtins, rest is kind of flexible like upper / lower, but consistent across scripts.

cyplo replied:
I see your comment above now - will change.

cyplo commented:
> When we are at it, can we add the `#[allow(clippy)]` to `tests.rs`? Because all these changes are here because of the `clippy` ...

added plus all the other changes - I think all should be consistent now across all the scripts, changed fuzzer script as well to be consistent with the rest @zrzka

cyplo commented:
yeah will do fo `temen` no worries :)
